io.openems.edge.controller.api.mqtt/src/io/openems/edge/controller/api/mqtt/Config.java

```diff
@@ -31,6 +31,18 @@
 
 	@AttributeDefinition(name = "Uri", description = "The connection Uri to MQTT broker.")
 	String uri() default "tcp://localhost:1883";
+	
+	@AttributeDefinition(name = "Certificate", description = "The path to the client certificate")
+	String certPath();
```

Comment:
Would it be possible or feasible to set the certificates directly as text here? In productive OpenEMS Edge distributions (like FENECON FEMS) the user is not allowed to upload (text)files to the edge device. If it was a direct configuration setting, the user could set it via OpenEMS UI.

Reply:
Thanks for the input. The private key and certificates can now be pasted as strings.